pkgs/sketch_pad/lib/main.dart

```diff
@@ -100,12 +102,11 @@ class DartPadMainPage extends StatefulWidget {
   final String? sampleId;
   final String? gistId;
 
-  const DartPadMainPage({
+  DartPadMainPage({
     required this.title,
     this.sampleId,
     this.gistId,
-    super.key,
-  });
+  }) : super(key: ValueKey('sample:$sampleId gist:$gistId'));
```

Comment:
This tells the framework to create a new DartPadMainPage whenever the gist or sample changes. This creates a new instance of `_DartPadMainPageState` whenever this happens. 

Alternatively, we could use the same key every time and use [didUpdateWidget](https://api.flutter.dev/flutter/widgets/State/didUpdateWidget.html) and call performInitialLoad again.